### DISCUSSION TITLE
Product quantity in decimal.

### DISCUSSION THREAD

**Discussion by naveed-alam** [woocommerce/woocommerce#45015] [WooCommerce, Blocks, and Site Editing]
I have products with quantity in decimal but cart page does not support it.

**Comment by opr** [comment]
Thanks for posting @naveed-alam - to clarify: your store allows users to purchase in quantities that are not whole numbers.

Do you have any information on how this is currently achieved, e.g. which hooks are you using to do this?

**Comment by naveed-alam** [comment]
add_filter( 'woocommerce_store_api_product_quantity_minimum', array(__CLASS__, 'minimum_quantity_limit'), 999, 3 );
add_filter( 'woocommerce_store_api_product_quantity_maximum', array(__CLASS__, 'maximum_quantity_limit'), 999, 3 );
add_filter( 'woocommerce_store_api_product_quantity_multiple_of', array(__CLASS__, 'multiple_of_quantity_limit'), 999, 3 );
I used these filters which work fine, but when the step(multiple of) is in decimal then it stop working.

**Reply by opr** [reply]
Thanks a lot for the extra info! Could you share what you're using _currently_ to get the decimal quantity working, assuming you have it working in the core (shortcode) cart/checkout flow?

**Comment by naveed-alam** [comment]
There is a file in WooCommerce plugin path: woocommerce/packages/woocommerce-blocks/src/StoreApi/Routes/V1/CartUpdateItem.php
public function get_args() {
		return [
			[
				'methods'             => \WP_REST_Server::CREATABLE,
				'callback'            => [ $this, 'get_response' ],
				'permission_callback' => '__return_true',
				'args'                => [
					'key'      => [
						'description' => __( 'Unique identifier (key) for the cart item to update.', 'woocommerce' ),
						'type'        => 'string',
					],
					'quantity' => [
						'description' => __( 'New quantity of the item in the cart.', 'woocommerce' ),
						'type'        => 'integer',// it should be float
					],
				],
			],
			'schema'      => [ $this->schema, 'get_public_item_schema' ],
			'allow_batch' => [ 'v1' => true ],
		];
	}
	integer should be float
